batch/batch/log_store.py

```diff
@@ -11,50 +11,76 @@
 
 
 class LogStore:
```

Comment:
This isn't really a LogStore, its something like "GCS API" because it deals with specs and logs. When I first saw this file, I went hunting for the `SpecStore` because I assumed such a class must exist since are specs are also stored in GCS.

Reply:
We could also have a LogStore and a SpecStore, 🤷 . I don't see two classes or one as obviously better or worse, as long as we use a descriptive name.